api-service/src/controllers/DataIngestion/DataIngestionController.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response } from "express";
+import * as _ from 'lodash';
+import dataIngestorSchema from "./DataIngestion.json";
+import { schemaValidation } from "../../services/ValidationService";
+import { ResponseHandler } from "../../helpers/ResponseHandler";
+import { send } from "../../connections/kafkaConnection";
+import { getDataset } from "../../services/DatasetService";
+
+const errorObject = {
+    invalidParam: {
+        "message": "datasetId parameter in url cannot be empty",
+        "statusCode": 400,
+        "errCode": "BAD_REQUEST"
+    },
+    invalidRequestBody: {
+        "message": "Invalid dataset config provided for ingestion",
+        "statusCode": 400,
+        "errCode": "BAD_REQUEST"
+    }
+}
+
+const dataIn = async (req: Request, res: Response) => {
+    try {
+        const datasetId = getDatasetId(req);
+        schemaValidation(req.body, dataIngestorSchema)
+        const dataset = await getDataset(datasetId)
+        const validData = await validation(req.body?.data, datasetId);
```

Comment:
if(! _.get(dataset, "dataValues.dataset_config.entry_topic")) {
 throw "Error entry topic is not defined"
}

Reply:
done